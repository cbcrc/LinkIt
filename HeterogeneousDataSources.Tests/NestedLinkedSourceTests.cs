using System.Collections.Generic;
using System.Linq;
using ApprovalTests.Reporters;
using NUnit.Framework;
using RC.Testing;

namespace HeterogeneousDataSources.Tests {
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class NestedLinkedSourceTests
    {
        [Test]
        public void LoadLink_NestedLinkedSource()
        {
            var loadLinkExpressions = new List<ILoadLinkExpression>{
                };

            var sut = new LoadLinkProtocol(
                TestUtil.ReferenceTypeConfigs,
                loadLinkExpressions
            );
            var content = new NestedContent{
                Id = 1,
                PersonId = "john"
            };
            var contentLinkedSource = new NestedLinkedSource{ Model = content };

            sut.LoadLink(contentLinkedSource);

            ApprovalsExt.VerifyPublicProperties(contentLinkedSource);
        }
    }


    public class NestedLinkedSource {
        public NestedContent Model { get; set; }
        public PersonLinkedSource Person { get; set; }
    }

    public class NestedContent {
        public int Id { get; set; }
        public string PersonId { get; set; }
    }

    public class PersonLinkedSource
    {
        public Person Model { get; set; }
        public Image SummaryImage{ get; set; }
    }
}
