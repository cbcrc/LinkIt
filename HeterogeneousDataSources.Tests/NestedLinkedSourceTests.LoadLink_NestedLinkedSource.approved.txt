{
  "model": {
    "id": 1,
    "personId": "john"
  },
  "person": null
}